itests/src/test/java/fi/metatavu/edelphi/test/mock/QueryMocker.java

```diff
@@ -0,0 +1,272 @@
+
+  public QueryMocker mock() {
+    return this;
+  }
+
+  public QueryMocker cleanup() {
+    for (Long pageSettingId : pageSettingIds) {
+      deleteQueryPageSetting(pageSettingId);
+    }
+
+    for (PageMeta page : pages) {
+      deleteQueryPage(page.getId());
+    }
+
+    for (Long sectionId : sectionIds) {
+      deleteQuerySection(sectionId);
+    }
+
+    for (Long queryId : queryIds) {
+      deleteQuery(queryId);
+    }
+
+    return this;
+  }
+
+  public long addQuery(String name, String urlName, Long parentFolderId, boolean visible, String description,
+      Integer indexNumber, boolean allowEditReply, Date closes, String state) {
+    Date now = new Date();
+    Long creatorId = 1l;
+    return createQuery(name, urlName, parentFolderId, visible, creatorId, now, creatorId, now, description, indexNumber,
+        allowEditReply, closes, state);
+  }
+
+  public long addDefaultSection(Long queryId) {
+    Date now = new Date();
+    Long creatorId = 1l;
+    return createQuerySection(queryId, "default", creatorId, now, creatorId, now, true, 0, true, true);
+  }
+
+  public long addSection(Long queryId, String title, boolean visible, Integer sectionNumber, boolean commentable,
+      boolean viewDiscussions) {
+    Date now = new Date();
+    Long creatorId = 1l;
+    return createQuerySection(queryId, title, creatorId, now, creatorId, now, visible, sectionNumber, commentable,
+        viewDiscussions);
+  }
+
+  public long addPage(Integer pageNumber, Long queryId, Long sectionId, String templateName) {
+    return mockPage(pageNumber, queryId, sectionId, readQueryPageMock(templateName));
+  }
+
+  private long mockPage(Integer pageNumber, Long queryId, Long sectionId, QueryPageMock pageMock) {
+    String type = pageMock.getType();
+    String title = pageMock.getTitle();
+    Boolean visible = pageMock.getVisible();
+    Date now = new Date();
+    Long creatorId = 1l;
+    long id = createQueryPage(pageNumber, queryId, creatorId, now, creatorId, now, type, title, sectionId, visible);
+    Map<String, SettingType> settingTypes = pageMock.getSettingTypes();
+
+    for (Entry<String, String> settingEntry : pageMock.getSettings().entrySet()) {
+      String settingKey = settingEntry.getKey();
+      String settingValue = settingEntry.getValue();
+      SettingType settingType = SettingType.TEXT;
+
+      if (settingTypes != null && settingTypes.containsKey(settingKey)) {
+        settingType = settingTypes.get(settingKey);
+      }
+
+      if (settingType == SettingType.QUERY_PAGE_IDS_BY_TYPE) {
+        settingValue = StringUtils.join(getPageIdsByType(settingValue), ",");
+      }
+
+      createQueryPageSetting(id, settingKey, settingValue);
+    }
+
+    return id;
+  }
+
+  private long createQueryPageSetting(Long queryPageId, String key, String value) {
+    long id = getNextId("QueryPageSetting");
+    String sql = "INSERT INTO " + "  QueryPageSetting (id, key_id, queryPage_id, value) " + "VALUES "
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "VALUES " 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "INSERT INTO " 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)